### ISSUE TITLE
Lab 6 Level 3 List

### ISSUE THREAD

**Issue by zhiqitan** [nus-cs2030/2021-s1#271]
Hi, any suggestion for handling the list of storing the Logger history? How to pass the elements inside the current list to the list of new Logger that created after calling map method? Am facing this issue which I think is because of make method return Logger instead of LoggerImpl (where the list is)? 
<img width="482" alt="Screenshot 2020-11-04 at 8 14 32 PM" src="https://user-images.githubusercontent.com/67494720/98110650-6029fb80-1eda-11eb-9682-750bad50da9e.png">
<img width="582" alt="Screenshot 2020-11-04 at 8 15 09 PM" src="https://user-images.githubusercontent.com/67494720/98110732-7b950680-1eda-11eb-8fdc-5ecee3d990fb.png">

**Comment by KeithCZW** [comment]
I think instead of calling the Logger.make function to create a Logger, just create a new LoggerImpl since it's the one that holds the list.

**Comment by sharleneq** [comment]
I think in terms of handling the log, add all the elements in the previous list and the new element after function is applied into the newLog. The return type for this map method could be changed from Logger<U> to LoggerImpl<U>, and like mentioned above, return a new LoggerImpl<U> with the newLog. Hope this helps!

**Comment by merylseow** [comment]
Like what Keith said, create a new `LoggerImpl` that passes `func.apply(this.value)` into the constructor. Also, you can't do `newLog.list` as list has private access so you'll probably need a getList() method if you need to access the list.

**Comment by itmeruoxin** [comment]
Yea your Logger.make probably creates a Logger that only has the initialising log statement. I think you can store the mapped T value as a T variable first, use a getter to get your current List of logs then add the new line of log to the list and use these new value and and logs list to create and return a new LoggerImpl.

**Comment by darrenhoon** [comment]
you could do a void method to specifically add items into a list for the new loggerImpl eg
```
void add(String message) {
    this.list.add(message);
}
```

**Comment by roopaniedu** [comment]
Yep I agree with the comments that the others gave. Note that your list might actually be private and/or final so its a good idea to add a method that allows some kind of manipulation of the list.
